File: src/brs/http/GetGuaranteedBalance.java
package brs.http;

import brs.Account;
import brs.BurstException;
import brs.services.ParameterService;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.Parameters.ACCOUNT_PARAMETER;
import static brs.http.common.Parameters.NUMBER_OF_CONFIRMATIONS_PARAMETER;
import static brs.http.common.ResultFields.GUARANTEED_BALANCE_NQT_RESPONSE;

;

final class GetGuaranteedBalance extends APIServlet.APIRequestHandler {

  private final ParameterService parameterService;

  GetGuaranteedBalance(ParameterService parameterService) {
    super(new APITag[] {APITag.ACCOUNTS, APITag.FORGING}, ACCOUNT_PARAMETER, NUMBER_OF_CONFIRMATIONS_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  JsonElement processRequest(HttpServletRequest req) throws BurstException {

    Account account = parameterService.getAccount(req);
    int numberOfConfirmations = parameterService.getNumberOfConfirmations(req);

    JsonObject response = new JsonObject();
    if (account == null) {
      response.addProperty(GUARANTEED_BALANCE_NQT_RESPONSE, "0");
    }
    else {
      response.addProperty(GUARANTEED_BALANCE_NQT_RESPONSE, String.valueOf(account.getBalanceNQT()));
    }

    return response;
  }

}

File: src/main/java/brs/http/GetGuaranteedBalance.kt
package brs.http

import brs.http.common.Parameters.ACCOUNT_PARAMETER
import brs.http.common.Parameters.NUMBER_OF_CONFIRMATIONS_PARAMETER
import brs.http.common.ResultFields.GUARANTEED_BALANCE_NQT_RESPONSE
import brs.services.ParameterService
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest


@Deprecated("This call is superseded by GetBalance which does what this does and more.")
internal class GetGuaranteedBalance @Deprecated("")
internal constructor(private val parameterService: ParameterService) : APIServlet.JsonRequestHandler(arrayOf(APITag.ACCOUNTS), ACCOUNT_PARAMETER, NUMBER_OF_CONFIRMATIONS_PARAMETER) {

    override suspend fun processRequest(request: HttpServletRequest): JsonElement {
        val account = parameterService.getAccount(request)
        val response = JsonObject()
        if (account == null) {
            response.addProperty(GUARANTEED_BALANCE_NQT_RESPONSE, "0")
        } else {
            response.addProperty(GUARANTEED_BALANCE_NQT_RESPONSE, account.balanceNQT.toString())
        }
        return response
    }

}